### DISCUSSION TITLE
Anyway to test run Goal/Penalty Animations?

### DISCUSSION THREAD

**Discussion by CamR17** [riffnshred/nhl-led-scoreboard#410] [General]
Hey Hey! 

First off, thanks so much for putting this together. I've had a ball getting my board up and running just in time for my Canes to take on the Golden Knights in the Cup Finals (Yes, I'm still in denial....).

Anyhow, I've been adding in some custom goal and penalty animations, but haven't found a way to preview them on the board to make sure they look okay/are legible. Is there a simple command I can use to run the .gifs?

Any tips? Or do I just need to wait until next season starts?

Thanks!

**Comment by falkyre** [comment]
@riffnshred can confirm on penalties, but I think you can only test the goal animations.  What you need to do is run the scoreboard manually (ie don't use supervisor) and add this to the command line when you run it.

`--test-goal-animation`

So your command line would look something like this:

`sudo python3 src/main.py --led-gpio-mapping=adafruit-hat --led-brightness=60 --led-slowdown-gpio=2 --test-goal-animation`

**Reply by CamR17** [reply]
Gave it a shot, but received the following error message:

```
usage: main.py [-h] [--testScChampions TESTSCCHAMPIONS]
               [--test-goal-animation TEST_GOAL_ANIMATION]
               [--testing-mode TESTING_MODE] [--led-rows LED_ROWS]
               [--led-cols LED_COLS] [--led-chain LED_CHAIN]
               [--led-parallel LED_PARALLEL] [--led-pwm-bits LED_PWM_BITS]
               [--led-brightness LED_BRIGHTNESS]
               [--led-gpio-mapping {regular,adafruit-hat,adafruit-hat-pwm}]
               [--led-scan-mode {0,1}]
               [--led-pwm-lsb-nanoseconds LED_PWM_LSB_NANOSECONDS]
               [--led-pwm-dither-bits LED_PWM_DITHER_BITS]
               [--led-show-refresh] [--led-slowdown-gpio {0,1,2,3,4}]
               [--led-limit-refresh LED_LIMIT_REFRESH]
               [--led-no-hardware-pulse LED_NO_HARDWARE_PULSE]
               [--led-rgb-sequence LED_RGB_SEQUENCE]
               [--led-pixel-mapper LED_PIXEL_MAPPER]
               [--led-row-addr-type {0,1,2,3,4}]
               [--led-multiplexing LED_MULTIPLEXING]
               [--led-panel-type LED_PANEL_TYPE]
               [--terminal-mode TERMINAL_MODE] [--updatecheck]
               [--updaterepo UPDATEREPO] [--ghtoken GHTOKEN] [--logcolor]
               [--loglevel LOGLEVEL]
main.py: error: argument --test-goal-animation: expected one argument
```

Do you think adding new animations to the preferred and opposing goal files causes this?

**Comment by riffnshred** [comment]
Im not home right now to confirm but try --test-goal-animation=true

**Reply by CamR17** [reply]
Thanks @riffnshred, That did the trick. 

I had to rename all the gifs I wanted to test to "goal_light_animation.gif" so the test would run them, but it was worth the effort. 

Thanks for the tips and putting this whole thing together!

**Comment by falkyre** [comment]
@riffnshred Maybe we should have it default to true when it's added to the command line?  You currently have it set to default=None.  Or change it something like this so it's always true even if you don't add =true:

`parser.add_argument("--logcolor", action="store_true", help="Display log in color (command line only)")`
